src/errors/validation_exception.rs

```diff
@@ -28,23 +29,45 @@ use super::ValError;
 #[derive(Clone)]
 #[cfg_attr(debug_assertions, derive(Debug))]
 pub struct ValidationError {
-    line_errors: Vec<PyLineError>,
+    line_errors: Vec<Arc<PyLineError>>,
+    validation_errors: Vec<Arc<ValidationError>>,
```

Comment:
A big question here is do we use `Arc` or `Py`. The former is better if most of the work is happening in Rust but requires more cloans, the latter requires the GIL and more Python interaction (slower?) but lets us pass around the objects more freely I think.

Reply:
why do we need either?

Reply:
Recursive struct?

Reply:
It's fine because it's inside a `Vec`, so it has to go on the heap anyway.

If it wasn't in a `Vec`, I would use a `Box`.